service/src/main/java/bio/terra/workspace/db/ResourceDao.java

```diff
@@ -57,46 +60,47 @@ public class ResourceDao {
-            .managedBy(
-                Optional.ofNullable(rs.getString("managed_by"))
-                    .map(ManagedByType::fromSql)
-                    .orElse(null))
-            .applicationId(Optional.ofNullable(rs.getString("associated_app")).orElse(null))
-            .assignedUser(rs.getString("assigned_user"))
-            .privateResourceState(
-                Optional.ofNullable(rs.getString("private_resource_state"))
-                    .map(PrivateResourceState::fromSql)
-                    .orElse(null))
-            .resourceLineage(
-                Optional.ofNullable(rs.getString("resource_lineage"))
-                    .map(
-                        resourceLineage ->
-                            DbSerDes.fromJson(
-                                resourceLineage,
-                                new TypeReference<List<ResourceLineageEntry>>() {}))
-                    .orElse(null));
-      };
+      (rs, rowNum) ->
+          new DbResource()
+              .workspaceUuid(UUID.fromString(rs.getString("workspace_id")))
+              .cloudPlatform(CloudPlatform.fromSql(rs.getString("cloud_platform")))
+              .resourceId(UUID.fromString(rs.getString("resource_id")))
+              .name(rs.getString("name"))
+              .description(rs.getString("description"))
+              .stewardshipType(fromSql(rs.getString("stewardship_type")))
+              .cloudResourceType(WsmResourceFamily.fromSql(rs.getString("resource_type")))
+              .resourceType(WsmResourceType.fromSql(rs.getString("exact_resource_type")))
+              .cloningInstructions(
+                  CloningInstructions.fromSql(rs.getString("cloning_instructions")))
+              .attributes(rs.getString("attributes"))
+              .accessScope(
+                  Optional.ofNullable(rs.getString("access_scope"))
+                      .map(AccessScopeType::fromSql)
+                      .orElse(null))
+              .managedBy(
+                  Optional.ofNullable(rs.getString("managed_by"))
+                      .map(ManagedByType::fromSql)
+                      .orElse(null))
+              .applicationId(Optional.ofNullable(rs.getString("associated_app")).orElse(null))
+              .assignedUser(rs.getString("assigned_user"))
+              .privateResourceState(
+                  Optional.ofNullable(rs.getString("private_resource_state"))
+                      .map(PrivateResourceState::fromSql)
+                      .orElse(null))
+              .resourceLineage(
+                  Optional.ofNullable(rs.getString("resource_lineage"))
+                      .map(
+                          resourceLineage ->
+                              DbSerDes.fromJson(
+                                  resourceLineage,
+                                  new TypeReference<List<ResourceLineageEntry>>() {}))
+                      .orElse(null))
+              .properties(DbSerDes.jsonToProperties(rs.getString("properties")));
```

Comment:
There should be Optional, to be consistent with workspace properties

```
              .properties(
                  Optional.ofNullable(rs.getString("properties"))
                      .map(DbSerDes::jsonToProperties)
                      .orElse(null))
```

Reply:
they can't be null.